### ISSUE TITLE
Combining view.data() with Angular's ng-repeat - behaviour change between 1.1.0 and 1.2.3

### ISSUE THREAD

**Issue by robguthrie** [techfort/LokiJS#131]
I've just experimented with a version change of Loki within our project and I've noticed a change in behaviour which is a show stopper for us.

With v1.1.0 I was able to build a loki view, add a function to the scope which returns the view's data, and ng-repeat over the function:

Angular controller
````
view = collection.addDynamicView('something')
$scope.records = function() { view.data() }
````

View template
````
<li ng-repeat="record in records()"></li>
````
This works really well with v1.1.0, but if I use v1.2.3 then I get a warning from angular that the digest cycle is stuck in a loop:

(sorry for the horrible link)
https://docs.angularjs.org/error/$rootScope/infdig?p0=10&p1=%5B%5B%7B%22msg%22:%22fn:%20regularInterceptedExpression%22,%22newVal%22:202,%22oldVal%22:199%7D%5D,%5B%7B%22msg%22:%22fn:%20regularInterceptedExpression%22,%22newVal%22:205,%22oldVal%22:202%7D%5D,%5B%7B%22msg%22:%22fn:%20regularInterceptedExpression%22,%22newVal%22:208,%22oldVal%22:205%7D%5D,%5B%7B%22msg%22:%22fn:%20regularInterceptedExpression%22,%22newVal%22:211,%22oldVal%22:208%7D%5D,%5B%7B%22msg%22:%22fn:%20regularInterceptedExpression%22,%22newVal%22:214,%22oldVal%22:211%7D%5D%5D

I would love to learn what is going on here.. Right now I can't see a way to bind the results from Loki 1.2.3 to angular's scope without this happening.

Thanks for your help!

**Comment by techfort** [comment]
Hi Robert,

I will investigate this asap. I have stumbled upon that problem with angular in a number of situations, the bottom line is that for some reason angular marks a certain property as dirty at each cycle, so it keeps trying to re-render the view, does this for 10 times then throws and error and stops.
It may be a problem with Loki, but there is a chance it's either angular or a slight implementation problem. If you had a fiddle or plunkr that would be great, in any case i'll be onto this tomorrow!

**Comment by robguthrie** [comment]
Thanks for the quick response. I'll try to put together a plunkr now.

**Comment by robguthrie** [comment]
Are there public copies of loki 1.1 and 1.2.3 that I can reference? Having trouble finding some.

**Comment by robguthrie** [comment]
I'm having trouble reproducing it in a plunkr, sorry. I'll keep trying.
 
In my app, I've just found that the problem only happens when I use applySimpleSort on the view. If I comment out the sorting, then everything works as expected.

**Comment by obeliskos** [comment]
Is this dynamic view persistent?  Yes a good test case would be great.

Recently I modified dynamic views to incorporate an async 'sort phase'.  If you do 100 inserts and then stop processing, our async will kick in an sort.  So this sort is happening whether you retrieve .data() or not.  

I can revisit this and possibly change that behavior or add new options to 'opt in' to this behavior if that proves to be your issue.

I would prefer to try in a plunkr if possible, but you might try commenting out line 2297 (or the whole setTimeout) in the DynamicView.queueSortPhase() method to see if that fixes your issue... this will force postponing of sorts until a data() call.

**Comment by robguthrie** [comment]
Here ya go!

http://plnkr.co/edit/ELTYXy6aknu6RpKGjOBF?p=preview

**Comment by robguthrie** [comment]
@obeliskos hope that plnkr works for you. I notice that the console errors only happen when there is more than 10 items in the collection

**Comment by obeliskos** [comment]
I believe I have fixed this error.  This seems to have been a simple bug not requiring the mental gymnastics I thought I was going to have to do so I guess that's good :) 

Let me know your results, I have checked in new sources and minified which you will need to download from github.

**Comment by robguthrie** [comment]
Thanks @obeliskos! That has fixed the issue for me!

Are you going to be cutting a new version I can specify rather than master?

**Comment by obeliskos** [comment]
Great! 

I have occasionally sifted through commit history and download older versions... version labeling 'checkpoints' might make it easier to identify.

I agree it would be nice to have more accessible historical versions.  @techfort has been handling versioning, mostly based on npm releases.  He might be able to recommend a specific versioning strategy.  He is out of timezone for now, but perhaps tomorrow.

**Comment by techfort** [comment]
@robguthrie @obeliskos thanks for working this out while I was asleep :) I published an npm 1.2.4 and tagged it as such so it should be available on bower as #1.2.4
Let me know if you have any issues with this. Bear in mind I was halfway through developing a feature, all tests and builds pass but don't go experimenting. Next release will most likely be 1.3.0, once external changes and unique/exact constraints are implemented.

**Comment by robguthrie** [comment]
Thanks again. 1.2.4 seems stable for my needs.
